tests/common/mod.rs

```diff
@@ -0,0 +1,188 @@
+use bitcoincore_rpc::{bitcoin::Network, json, RpcApi};
+use bitcoind::{BitcoinD, Conf, ConnectParams};
+use flate2::read::GzDecoder;
+use std::env;
+use std::process::{Child, Command, Stdio};
+use std::thread;
+use tar::Archive;
+use tempfile::{tempdir, Builder, TempDir};
+use tokio::time::Duration;
+
+const LND_VERSION: &str = "0.16.0";
+
+// setup_test_infrastructure spins up all of the infrastructure we need to test LNDK, including a bitcoind node and two
+// LND nodes. LNDK can then use this test environment to run.
+pub async fn setup_test_infrastructure() -> (BitcoinD, LndNode, TempDir) {
+    let (bitcoind, bitcoind_dir) = setup_bitcoind().await;
+    let lnd_exe_dir = download_lnd().await;
+
+    let lnd_node = LndNode::new(bitcoind.params.clone(), lnd_exe_dir);
+    lnd_node.setup_client().await;
+
+    return (bitcoind, lnd_node, bitcoind_dir);
+}
+
+pub async fn setup_bitcoind() -> (BitcoinD, TempDir) {
+    let bitcoind_dir = tempdir().unwrap();
+    let bitcoind_dir_path = bitcoind_dir.path().clone().to_path_buf();
+    let mut conf = Conf::default();
+    conf.tmpdir = Some(bitcoind_dir_path);
+    conf.args = vec![
+        "-regtest",
+        "-fallbackfee=0.0001",
+        "-zmqpubrawblock=tcp://127.0.0.1:28332",
+        "-zmqpubrawtx=tcp://127.0.0.1:28333",
+    ];
+    let bitcoind = BitcoinD::from_downloaded_with_conf(&conf).unwrap();
+
+    // Mine 101 blocks in our little regtest network so that the funds are spendable.
+    // (See https://bitcoin.stackexchange.com/questions/1991/what-is-the-block-maturation-time)
+    let address = bitcoind
+        .client
+        .get_new_address(None, Some(json::AddressType::Bech32))
+        .unwrap();
+    let address = address.require_network(Network::Regtest).unwrap();
+    bitcoind.client.generate_to_address(101, &address).unwrap();
+
+    (bitcoind, bitcoind_dir)
+}
+
+#[cfg(all(target_os = "linux", target_arch = "x86_64"))]
+fn lnd_download_filename() -> String {
+    format!("lnd-linux-amd64-v{}-beta.tar.gz", &LND_VERSION)
+}
```

Comment:
Could you add mac support as well :')

Reply:
Absolutely :)

Reply:
Hey @carlaKC question for you on this. I'm not super familiar with running LND on a Mac, which of the premade LND binaries is compatible with it? https://github.com/lightningnetwork/lnd/releases/tag/v0.16.2-beta

Reply:
`darwin-arm64` will work for apple chips. 

May be worth failing the test if it's not running on one of the supported OS-es? New folks can add as required, but I think that an explicit failure would be good.

Reply:
Good idea :+1: